js/forms.js

```diff
@@ -279,85 +279,181 @@
      * Auto complete plugin  *
      *************************/
     $.fn.autocomplete = function (options) {
-      // Defaults
       var defaults = {
-        data: {}
-      };
+        data: {},
+        maxElementsAmount: null
+      },
+      currentLi = 0,
+      autocompleteOption = null,
+      currentAutocompleteVal = '';
 
       options = $.extend(defaults, options);
 
-      return this.each(function() {
-        var $input = $(this);
-        var data = options.data,
-            $inputDiv = $input.closest('.input-field'); // Div to append on
+      // Set input value
```

Comment:
Comment maybe doesn't bring so much, but I do that because it's easier for me to read a code. Comments have different color and for me when I looking through the code it's like headers.

I separated that intentionally. First and less important I think right now it's just more readable. In one section you define functions - in different section you fired that. 
But more important thing: if you move everything to one loop and have for example 100 autocomplete inputs, when you run this loop you will define all this function 100 times. Right now you will define functions only once and you can fire that 100 times. It's just about performance.

And you cannot move this code to line 450 because this function just will not work.